File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/factory/transport/TransportFactory.java
// TransportFactory.java

package com.bigboxcode.designpattern.factory.transport;

public class TransportFactory {
    public Transport getTransport(String type) {
        if (type == null) {
            return null;
        }

        if (type.equalsIgnoreCase("bike")) {
            return new Bike();
        }

        if (type.equalsIgnoreCase("car")) {
            return new Car();
        }

        if (type.equalsIgnoreCase("plane")) {
            return new Plane();
        }

        return null;
    }
}

File: design-patterns/ts/factory/transport/transport-factory.ts
// transport-factory.ts

import Transport from "./transport";
import Bike from "./bike";
import Car from "./car";
import Plane from "./plane";

class TransportFactory {

    // Make this method static if needed
    getTransport(type: string): (Transport | null) {
        if (type.toLocaleLowerCase() === "bike") {
            return new Bike();
        }

        if (type.toLocaleLowerCase() === "car") {
            return new Car();
        }

        if (type.toLocaleLowerCase() === "plane") {
            return new Plane();
        }

        return null;
    }
}

export default TransportFactory;